File: argos-java/argos-core/src/main/kotlin/argos/core/assertion/IntentAssertion.kt
package argos.core.assertion

import argos.api.*
import gaia.sdk.api.skill.SkillEvaluation
import gaia.sdk.core.Gaia
import io.reactivex.Flowable
import org.reactivestreams.Publisher

/**
 * This IAssertion implementation is used to compare the result of the
 * intent detection skill evaluation with an expected result value.
 */
class IntentAssertion(val spec: IntentAssertionSpec) : IAssertion {

    override fun assert(options: ArgosOptions): Publisher<IAssertionResult> {
        val gaiaRef = Gaia.connect(options.config)

        val request: Publisher<SkillEvaluation> = gaiaRef.skill(options.config.url)
                .evaluate(mapOf("text" to spec.text, "treshold" to spec.score))

        return Flowable.fromPublisher(request)
                .map { it.asMap() }
                .map { e ->
                    if (e[":type"] == "Match" && e["reference"] == spec.intent)
                        Success("success")
                    else
                        Failure("failure")
                }
    }

}

File: argos-python/argos/core/assertion/IntentAssertion.py
from dataclasses import dataclass

from rx import Observable

from argos.api import IAssertion, ArgosOptions, IAssertionResult


@dataclass
class IntentAssertionSpec:
    text:str
    intent:str
    score:float = 0.85


@dataclass
class IntentAssertion(IAssertion):
    spec:IntentAssertionSpec

    def assertion(self, options:ArgosOptions) -> Observable[IAssertionResult]:
        pass